src/main/java/org/jabref/gui/groups/GroupTreeView.java

```diff
@@ -220,7 +223,7 @@ public void initialize() {
                     event.acceptTransferModes(TransferMode.MOVE, TransferMode.LINK);
 
                     //expand node and all children on drag over
-                    row.getTreeItem().setExpanded(true);
+                    row.getTreeItem().setExpanded(previousDragOverTreeItem.expandGroup(row.getTreeItem()));
```

Comment:
Please move the call to `row.getTreeItem().setExpanded` also to the `expandGroup` method (so that it does what it's name suggests).